### ISSUE TITLE
`webContent.print` fail because "No valid printers available"

### ISSUE THREAD

**Issue by abnerlee** [electron/electron#24664]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

When I run `webContents.print` (https://www.electronjs.org/docs/api/web-contents#contentsprintoptions-callback):

```js
webContents.print({silent:false}, (success, failureReason) => {
  // success will be false
 // failureReason will be "No valid printers available"
});
```
Instead of popping up a print setting dialog, the `callback` will return with `failureReason` of "No valid printers available".

In windows, it works fine.

* **Electron Version:**
  * 8.x / 9.1.1
* **Operating System:**
  * macOS 10.15.5
* **Last Known Working Electron version:**
  * 6.x

### Expected Behavior

popping up a print setting dialog

### Actual Behavior

the `callback` will return with `failureReason` of "No valid printers available".

### Additional Information

Windows version works fine

**Comment by codebytere** [comment]
This is expected behavior. Our print logic will not proceed unless your machine has a valid printer on the network configured.

**Comment by abnerlee** [comment]
@codebytere This should be the spec when `silent` is `true`, when `silent` is false, it should "ask user for print settings"

I can show the dialog on xcode and other apps even when there's no printers, because even when there's no printers, PDF, send in mail, etc should be supported.
 
<img width="439" alt="截屏2020-07-22 21 27 26" src="https://user-images.githubusercontent.com/1621242/88181989-323de580-cc62-11ea-9e6f-c535c38d205f.png">

**Comment by abnerlee** [comment]
sorry @codebytere,  would you confirm why electron uses a spec that differs from other apps?

**Comment by andsmedeiros** [comment]
I can confirm this on Ubuntu 20.04, Linux 5.4.0-45-generic and Electron 10.1.1.
From the devtools, calling `window.print()` opens the print dialog as expected. 
Calling `electron.remote.getCurrentWebContents().print()` fails silently. Logging the callback arguments yields "no valid printers available". There ARE available printers.
I will try it on Windows later.

**Comment by mdg1019** [comment]
I'm using Ubuntu 20.04 and Electron 10.1.2 and am experiencing the same issue.

**Comment by abnerlee** [comment]
-> #25498

**Comment by caffeinatedbits** [comment]
I ran into this same issue using a BrowserWindow which contains multiple BrowserViews on macOS. As a temporary fix, I used ipcMain to send a message via BrowserView.webContents.send("print") and, within the BrowserView, ipcRenderer to listen for that via ipcRenderer.on("print"). Then, within that listener, I call window.print(). Problem solved for now, though BrowserView.webContents.print() should do the same thing, which instead still returns the error mentioned ("no valid printers available") as of electron@12.0.1.
